
introduce -Wundef check to catch macro typos
more infos about motivation at https://sourceware.org/glibc/wiki/Wundef

TODO: better define mc_fhl_* as macros too

patch 02/04: remove superfluous macro check

Signed-off-by: Andreas Mohr <and@gmx.li>

--- a/lib/filehighlight/get-color.c
+++ b/lib/filehighlight/get-color.c
@@ -49,9 +49,6 @@
 inline static gboolean
 mc_fhl_is_file (file_entry_t * fe)
 {
-#if S_ISREG == 0
-    (void) fe;
-#endif
     return S_ISREG (fe->st.st_mode);
 }
 
@@ -64,18 +61,12 @@
 inline static gboolean
 mc_fhl_is_dir (file_entry_t * fe)
 {
-#if S_ISDIR == 0
-    (void) fe;
-#endif
     return S_ISDIR (fe->st.st_mode);
 }
 
 inline static gboolean
 mc_fhl_is_link (file_entry_t * fe)
 {
-#if S_ISLNK == 0
-    (void) fe;
-#endif
     return S_ISLNK (fe->st.st_mode);
 }
 
@@ -100,50 +91,33 @@
 inline static gboolean
 mc_fhl_is_device_char (file_entry_t * fe)
 {
-#if S_ISCHR == 0
-    (void) fe;
-#endif
     return S_ISCHR (fe->st.st_mode);
 }
 
 inline static gboolean
 mc_fhl_is_device_block (file_entry_t * fe)
 {
-#if S_ISBLK == 0
-    (void) fe;
-#endif
     return S_ISBLK (fe->st.st_mode);
 }
 
 inline static gboolean
 mc_fhl_is_special_socket (file_entry_t * fe)
 {
-#if S_ISSOCK == 0
-    (void) fe;
-#endif
     return S_ISSOCK (fe->st.st_mode);
 }
 
 inline static gboolean
 mc_fhl_is_special_fifo (file_entry_t * fe)
 {
-#if S_ISFIFO == 0
-    (void) fe;
-#endif
     return S_ISFIFO (fe->st.st_mode);
 }
 
 inline static gboolean
 mc_fhl_is_special_door (file_entry_t * fe)
 {
-#if S_ISDOOR == 0
-    (void) fe;
-#endif
-
     return S_ISDOOR (fe->st.st_mode);
 }
 
-
 inline static gboolean
 mc_fhl_is_special (file_entry_t * fe)
 {
